packages/remix-dev/vite/plugin.ts

```diff
@@ -450,6 +537,25 @@ export const remixVitePlugin: RemixVitePlugin = (remixUserConfig = {}) => {
 
   /** Mutates `ctx` as a side-effect */
   let updateRemixPluginContext = async (): Promise<void> => {
+    // Default config values are completely overridden by user/preset config,
+    // not merged. This means that our preset merging pipeline needs to start
+    // with an empty object. Conceptually this means that presets are operating
+    // on the user config object as it would be defined in your Vite config.
+    let accumulatedRemixConfigPresets: VitePluginConfig = {};
+
+    for (let preset of remixUserConfig.presets ?? []) {
+      let { remixConfig: getConfigPreset } = preset;
+      let remixConfigPreset =
+        getConfigPreset &&
+        omit(
+          await getConfigPreset({ remixConfig: accumulatedRemixConfigPresets }),
```

Comment:
I'm thinking I might pull this `remixConfig` arg from the preset since it's of limited value and arguably pretty confusing. Because it's the accumulator, it only has access to config values defined in earlier presets. The original intention of this arg was to support config validation, but this should now be handled in the `remixConfigResolved` hook since that contains the final config.

Reply:
Updated.